### ISSUE TITLE
Property setter wrongly required to be marked `mut` in static structs

### ISSUE THREAD

**Issue by disarray2077** [beefytech/Beef#1084]
Property setters are required to be `mut` but they raise a "unnecessary mut" warning when marked as `mut` when they are member of a static struct.

**Comment by bfiete** [comment]
I'm not sure I understand - what do you mean by "Property setters are required to be mut"? They are only required to be `mut` if they mutate a struct's `this`.

**Comment by bfiete** [comment]
Could you provide a code example where the property setter is required to be mut?

**Comment by disarray2077** [comment]
Maybe my assumption that the static block has basically the same context of a static struct was wrong and this only happens with static blocks, sorry if this is the case, here is a code where this problem occurs: 
```
namespace BeefBugRepros
{
    static
    {
        public static ref Event<Action> MyEvent
        {
            get;
            set; // error in this line
        } ~ _.Dispose();
    }
}
```

**Comment by bfiete** [comment]
Ah, got it. Fixed at 8796b3bcc158b51929e224ba91c88a9e98989a1b
